go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,388 @@
+	Client   amplitude.Client
+	mutex    sync.Mutex
+}
+
+// Load initializes the Ampli wrapper.
+// Call once when your application starts.
+func (a *Ampli) Load(options LoadOptions) {
+	a.mutex.Lock()
+	a.Disabled = options.Disabled
+	a.mutex.Unlock()
+
+	if a.Client != nil {
+		log.Default().Printf("Warn: Ampli is already initialized. Ampli.load() should be called once at application start up.")
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration(),
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Instance.APIKey or APIKey[options.Environment], " +
+			"or options.Instance.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration()
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration().Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+	return !a.Disabled && a.Client != nil
+}
+
+func (a *Ampli) setUserID(userID string, eventOptions *amplitude.EventOptions) {
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	a.setUserID(userID, &eventOptions)
+
+	baseEvent := event.ToEvent()
+	baseEvent.EventOptions = eventOptions
+
+	a.Client.Track(baseEvent)
+}
+
+// Identify identifies a user and set user properties.
+func (a *Ampli) Identify(userID string, identify *Identify, eventOptions amplitude.EventOptions) {
+	a.Track(userID, identify, eventOptions)
+}
+
+// GroupIdentify identifies a group and set group properties.
+func (a *Ampli) GroupIdentify(groupType string, groupName string, group *Group, eventOptions amplitude.EventOptions) {
+	group.groups = map[string][]string{groupType: {groupName}}
+	event := group.ToEvent()
+	event.EventOptions = eventOptions
+
+	a.Client.Track(event)
+}
+
+// SetGroup sets group for the current user.
+func (a *Ampli) SetGroup(userID string, groupType string, groupName []string, eventOptions amplitude.EventOptions) {
+	a.setUserID(userID, &eventOptions)
```

Comment:
I think it's better to move the setUserId logic into the core logic. 
cc: @justin-fiedler

Reply:
Agree. Our current interface `Amplitude.Identify(Identify, EventOptions)` doesn't require `userid`, which is optional to pass to EventOptions.

If a user run `client.identify(identify_obj, EventOptions())` it returns an invalid event error. 

It would be better if we could update identify interface to require users to pass userid `Amplitude.Identify(userid, Identify, EventOptions)

Reply:
This is how we do for kotlin. 
https://github.com/amplitude/Amplitude-Kotlin/pull/37/files
For the identify call we can overwrite the eventOptions.userId = userID. 
Then in the core SDK, we can call `setUserId` if there has eventOptions.userId exist in the identify function.

Reply:
Node.js (server side):
https://github.com/amplitude/ampli-examples/blob/main/node/typescript/v2/AmpliApp/src/ampli/index.ts#L603

Reply:
FYI:
This setUserId is just set the eventOptions.userId. 
And since this is the server-side SDK the logic is different.